### PR TITLE
Created Insertion_Sort.php

### LINKED ISSUE #2148: Insertion Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart
- [x] Go
- [x] Java
- [x] JS
- [x] Kotlin
- [ ] Php - @saloniankita 
- [x] Python
- [ ] Readme - @debdutgoswami
- [x] Ruby
- [x] Rust - @m-code12
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by saloniankita**
Fixes #2148  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- PHP
- 

#### Files Added:
- Insertion_Sort.php
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
please include the issue number in PR template

**Comment by saloniankita** [comment]
Done @asha15

**Comment by somya-kapoor** [comment]
please include the issue number in PR template, not the PR number

**Comment by saloniankita** [comment]
Is this okay now @somya-kapoor ?

File: Insertion_Sort/Insertion_Sort.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?php
+
+class InsertionSort 
+{
+    private $arr;
+    public function __construct($arr) 
+    {
+         $this->arr = $arr;
+             }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:9]: inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+    //Printing the array
+    public function printArray($array) 
+    {
+         for($i = 0; $i < count($array); $i++) 
+             {
+                  echo $array[$i]." ";
+                      }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:17]: inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    public function sort() 
+    {
+        for($i = 1; $i < count($this->arr); $i++) 
+            {
+                 $temp = $this->arr[$i];
+                 while($i > 0 && $temp < $this->arr[$i-1]) 
+                 {
+                      $this->arr[$i] = $this->arr[$i-1];
+                      --$i;
+                      }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:30]: inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    $is = new InsertionSort($num);
+    
+//Printing Unsorted array
+echo "<br>Unsorted Array : ";
+$is->printArray($num);
+    
+//Printing Sorted array
+echo "<br>Sorted Array : ";
+$is->printArray($is->sort());
+
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:57]: Remove this line

File: Insertion_Sort/Insertion_Sort.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+/*
+Implementing Insertion Sort Algorithm :
+
+Unsorted Array : 5 2 3 4 8 6 1 7
+
+Sorted Array : 1 2 3 4 5 6 7 8
+*/
+
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:66]: remove this line

**Comment by saloniankita** [comment]
Is it okay now @somya-kapoor ?

File: Insertion_Sort/Insertion_Sort.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+class InsertionSort 
+{
+    private $arr;
+    public function __construct($arr) 
+    {
+         $this->arr = $arr;
+      }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:9]: Inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    
+    //Printing the array
+    public function printArray($array) 
+    {
+         for($i = 0; $i < count($array); $i++)
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:14]: Use 4 space indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    public function sort() 
+    {
+        for($i = 1; $i < count($this->arr); $i++) 
+            {
+                 $temp = $this->arr[$i];
+                 while($i > 0 && $temp < $this->arr[$i-1]) 
+                 {
+                      $this->arr[$i] = $this->arr[$i-1];
+                      --$i;
+                   }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:30]: Inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php (hunk 5 above)
Comment (somya-kapoor) [reply Insertion_Sort/Insertion_Sort.php:65]: remove this line line 65

File: Insertion_Sort/Insertion_Sort.php [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+         $this->arr = $arr;
+      }
+    
+    //Printing the array
+    public function printArray($array) 
+    {
+         for($i = 0; $i < count($array); $i++) 
+             {
+                  echo $array[$i]." ";
+               }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:17]: Use 4 space indentation

**Comment by saloniankita** [comment]
@somya-kapoor Take a look now.

File: Insertion_Sort/Insertion_Sort.php [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        for($i = 1; $i < count($this->arr); $i++) 
+            {
+                $temp = $this->arr[$i];
+                while($i > 0 && $temp < $this->arr[$i-1]) 
+                {    
+                    $this->arr[$i] = $this->arr[$i-1];
+                    --$i;
+                }
+                    $this->arr[$i] = $temp;
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:31]: Inconsistency in indentation

**Comment by saloniankita** [comment]
Done @somya-kapoor

File: Insertion_Sort/Insertion_Sort.php [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    $is = new InsertionSort($num);
+    
+//Printing Unsorted array
+echo "<br>Unsorted Array : ";
+$is->printArray($num);
+  
+//Printing Sorted array
+echo "<br>Sorted Array : ";
+$is->printArray($is->sort());
+/*
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:57]: Leave a line before this

File: Insertion_Sort/Insertion_Sort.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                $temp = $this->arr[$i];
+                while($i > 0 && $temp < $this->arr[$i-1]) 
+                {    
+                    $this->arr[$i] = $this->arr[$i-1];
+                    --$i;
+                }
+                $this->arr[$i] = $temp;
+            }    
+            return $this->arr;
+    }
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:34]: Inconsistency in indentation

Comment (somya-kapoor) [reply Insertion_Sort/Insertion_Sort.php:34]: Inconsistency in indentation

File: Insertion_Sort/Insertion_Sort.php [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+                    $this->arr[$i] = $this->arr[$i-1];
+                    --$i;
+                }
+                $this->arr[$i] = $temp;
+            }    
+            return $this->arr;
+    }       
+}    
+ 
+
```
Comment (somya-kapoor) [inline Insertion_Sort/Insertion_Sort.php:37]: Remove this line

**Comment by saloniankita** [comment]
Take a look @somya-kapoor

**Comment by saloniankita** [comment]
Kindly review this PR @asha15 @somya-kapoor

File: Insertion_Sort.php [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+$is->printArray($is->sort());
+
+/*
+Implementing Insertion Sort Algorithm :
+
+Unsorted Array : 5 2 3 4 8 6 1 7
+
+Sorted Array : 1 2 3 4 5 6 7 8
+*/
+?>
```
Comment (singh-shreya6) [inline Insertion_Sort.php:64]: Please add a blank line at end of file.

**Pull request merged**